### DISCUSSION TITLE
Does it have to be global -g? Will it work non-global??

### DISCUSSION THREAD

**Discussion by ewebgh33** [FlowiseAI/Flowise#1411] [Q&A] [answered]
Hi

I am testing a range of local LLM apps and Flowise got a lot of upvotes on reddit when I asked about a good tool for local+RAG and configurability.

But all these tools all want a heck of a lot of dependences and I am wary of breaking some/all when I install a new one. So:

Background -
I am used to python environments where I can set a venv and have everything contained. NOTHING I do with python and LLMs is global.
My understanding is npm is similar, BUT
Installation recommendation looks to be global by default? 
`npm install -g flowise`

Any reason for global install? I really don't want to break something else I am using if Flowise installs something that conflicts.

Will it work if I did this instead?
`npm install flowise`

I see there is also a docker. I suppose I could run that, but I've already had problems with another local LLM tool (Cheshire Cat) where because the app is running in docker, but needs to connect to LLMs outside docker, it's a pain and a hassle. So I would prefer not to use docker, but also not to install a bunch of things globally if I can avoid it.

As an aside, my node --version also shows 21.1.0, is this also OK? 

Thanks

**Comment by ewebgh33** [comment]
Bump to this
I really don't understand the global thing
I guess I can just try non -g but would also be nice not to waste my time if this HAS to be global to install.

**Comment by Jaredude** [comment, marked as answer]
You would use -g when you are going to be using Flowise as a service running on a server. For example, you're putting Flowise on a server or docker container that's going to run your staging or production flows.

If you're doing dev to make changes to Flowise source code, you wouldn't install at all. Clone the repo and run local.

Omitting the -g option on an npm install would install flowise as a dependency in another project (you would need to have that project already set up). So, unless you're making a nodejs project that's going to be extending Flowise, use the -g option so that Flowise is installed globally and can be started as a system service on your server. Hope that helps.

**Comment by ewebgh33** [comment]
Thanks for the explanation.

It would not be running on a server, but neither do I really want to change it or do dev.
I just want to run it local for a single user, one a regular PC (with 4090s in it) to test for how it could help my small business (which does not use servers at all).

So I am wary of installing it globally, as I'm testing a range of other things and don't want to cause any conflicts by dumping a range of requirements outside a conda or other environment, that's all.

**Reply by HenryHengZJ** [reply]
installing it globally allows you to quickly get it up and running. As it's not a python project so you prob dont need to mess around with requirements and conda. If you install globally, everything will just get installed under a folder thats it. Here's mine - `C:\Users\Henry\AppData\Roaming\nvm\v19.8.1\node_modules\flowise`

Or like @Jaredude suggested, using Docker if you dont want to mess with other npm libraries on your local machine

**Comment by Jaredude** [comment]
Under that scenario running it using docker makes the most sense since that's the exact scenario docker was created for in the first place. It would still be a -g install on the docker container. The Flowise source has a docker compose file that will get things going quickly. Good luck.
